pandas/core/common.py

```diff
@@ -277,6 +277,11 @@ def maybe_iterable_to_list(obj: Union[Iterable[T], T]) -> Union[Collection[T], T
     """
     if isinstance(obj, abc.Iterable) and not isinstance(obj, abc.Sized):
         return list(obj)
+    # error: Incompatible return value type (got
+    # "Union[pandas.core.common.<subclass of "Iterable" and "Sized">,
+    # pandas.core.common.<subclass of "Iterable" and "Sized">1, T]", expected
+    # "Union[Collection[T], T]")  [return-value]
+    obj = cast(Collection, obj)
```

Comment:
our signature here is not entirely correct. If we pass an Iterable with a \_\_len__ method which is not necessarily a collection, the object is returned unchanged.

I think best to leave signature unchanged for now and could use Protocol in the future.

Reply:
since i'm lying to the type checker a #type: ignore is probably more preferable to a cast.